### PR TITLE
[CPM-118]  Regulation UI Edit

### PR DESCRIPTION
**Pull request by Richardng12**
- Redid regulation model without validation
- Implemented AND, OR buttons to build a regulation string
- Validation to be done in future (to avoid blockages)
- Create course button (back-end) done in future PR

### CODE REVIEW

File: client/src/pages/Courses/RegulationModalValidation.js [hunk 1]
```diff
@@ -0,0 +1,345 @@
...
+//     Switch,
+//     FormLabel,
+//     ButtonGroup,
+// } from "@chakra-ui/core";
+
+// const defaultReg = {
+//     type: "POINTS",
+//     points: 15,
+//     courses: [],
+// };
```
Comment (FocalChord) [inline client/src/pages/Courses/RegulationModalValidation.js:34]: Should the file be deleted?

**Review by vee-christella — COMMENTED** [review]
I found a weird bug + couple of comments 🤔 Can we try to add the courses via the `Enter` key instead of always clicking the down arrow button? Feels more intuitive that way!

To reproduce the bug:
1. Click to create a course
2. Add Prereqs
3. Switch to note mode
4. Type something in and click the add button
5. Close the dialog (don't save)
6. Re-enter the prereqs dialog
7. **The text from before is still there. But when you click plus, it says '15 points from [BLANK]' instead of the text**. See below:

image


Speaking of which, do we want the note to be '15 points from'? Should the note just be exactly what the user types in instead?

**Comment by daltonlim** [comment]
we also have the classic 
image

File: client/src/pages/Courses/RegulationModal.js [hunk 2]
```diff
@@ -93,27 +154,28 @@ const RegulationModal = ({ isOpen, onClose, updateRegs, title }) => {
...
                             {toggleInput ? (
                                 <>
                                     <Flex m="2">
                                         <Input id="courses" placeholder="Courses" onChange={handleCourseChange} value={courseTextValue} />
-                                        <IconButton aria-label="Add input" icon="arrow-down" onClick={handleCourseClick} />
+                                        <IconButton
+                                            aria-label="Add input"
+                                            icon="arrow-down"
+                                            onClick={handleCourseClick}
+                                            isDisabled={courseTextValue !== "" ? false : true}
```
Comment (daltonlim) [inline client/src/pages/Courses/RegulationModal.js:168]: could be simplified to courseTextValue === ""

File: client/src/pages/Courses/RegulationTable.js [hunk 3]
```diff
@@ -1,39 +1,101 @@
...
+                    backgroundColor="#162971"
+                    onClick={handleOpen}
+                    isDisabled={regulationType.length > 0 ? true : false}
+                >
                     <Text textAlign="center" color="white">
                         Add {name}
                     </Text>
                 </Button>
-                {allRegsArray.map((reg) => (
+                {regulationType.length > 0 ? (
```
Comment (daltonlim) [inline client/src/pages/Courses/RegulationTable.js:88]: Could be chnaged to && if second option is empty div

**Review by daltonlim — CHANGES_REQUESTED** [review]

File: client/src/pages/Courses/RegulationModal.js [hunk 4]
```diff
@@ -130,46 +192,93 @@ const RegulationModal = ({ isOpen, onClose, updateRegs, title }) => {
...
+                        </Stack>
+                    </ModalBody>
 
                     <ModalFooter>
-                        <Button variantColor="blue" mr={3} onClick={handleClose}>
+                        <Button variantColor="red" mr={3} onClick={handleClose}>
                             Close
                         </Button>
-                        <Button variant="ghost" onClick={handleConfirm}>
+                        <Button variantColor="green" onClick={handleConfirm} isDisabled={outputRegulation.length > 0 ? false : true}>
```
Comment (daltonlim) [inline client/src/pages/Courses/RegulationModal.js:281]: Could be simplified to isDisabled={outputRegulation.length}

File: client/src/pages/Courses/RegulationModal.js (hunk 2 above)
Comment (daltonlim) [inline client/src/pages/Courses/RegulationModal.js:168]: isDisabled={courseTextValue === "" }

Comment (FocalChord) [reply client/src/pages/Courses/RegulationModal.js:168]: Alternatively `isDisabled={!courseTextValue}` since empty string is falsey in JS

Comment (daltonlim) [reply client/src/pages/Courses/RegulationModal.js:168]: ^ use this. reads better

File: client/src/pages/Courses/RegulationModalValidation.js (hunk 1 above)
Comment (wryao64) [reply client/src/pages/Courses/RegulationModalValidation.js:34]: Nah, we want to keep it in case we decide to go do validation properly again

Comment (FocalChord) [reply client/src/pages/Courses/RegulationModalValidation.js:34]: Hm... I feel we shouldn't commit an unused file for the sake of keeping things clean. Should we make a gist of this and save it on discord if we ever need it in the future?

File: client/src/pages/Courses/RegulationModal.js (hunk 2 above)
Comment (Richardng12) [reply client/src/pages/Courses/RegulationModal.js]: Fixed

**Review by daltonlim — APPROVED** [review]
LGTM

**Review by FocalChord — APPROVED** [review]
Very nice sir

**Comment by Richardng12** [comment]
> 
> 
> I found a weird bug + couple of comments 🤔 Can we try to add the courses via the `Enter` key instead of always clicking the down arrow button? Feels more intuitive that way!
> 
> To reproduce the bug:
> 
>     1. Click to create a course
> 
>     2. Add Prereqs
> 
>     3. Switch to note mode
> 
>     4. Type something in and click the add button
> 
>     5. Close the dialog (don't save)
> 
>     6. Re-enter the prereqs dialog
> 
>     7. **The text from before is still there. But when you click plus, it says '15 points from [BLANK]' instead of the text**. See below:
> 
> 
> image
> 
> Speaking of which, do we want the note to be '15 points from'? Should the note just be exactly what the user types in instead?

Fixed

**Review by vee-christella — APPROVED** [review]
LGTM now 🍡  The only other consideration  might be to indicate that a AND or OR has to be clicked before adding another regulation (because atm it just disables with no indication). But we can do this in another ticket if you'd like -- if so, can one of you create a ticket for this?

**Comment by Richardng12** [comment]
> 
> 
> LGTM now 🍡 The only other consideration might be to indicate that a AND or OR has to be clicked before adding another regulation (because atm it just disables with no indication). But we can do this in another ticket if you'd like -- if so, can one of you create a ticket for this?

Like a hover thing over a disabled plus?

**Comment by vee-christella** [comment]
> Like a hover thing over a disabled plus?

What do you mean by this? I'm thinking maybe a red asterisk that pops up if they attempt to click `+` button before adding an AND or an OR statement

**Comment by Richardng12** [comment]
> 
> 
> > Like a hover thing over a disabled plus?
> 
> What do you mean by this? I'm thinking maybe a red asterisk that pops up next to the clauses if they attempt to click `+` button before adding an AND or an OR statement

I meant something like this:

image

But i realised tooltips don't work for disabled buttons, so may just have to have a red asterisk

**Comment by vee-christella** [comment]
> by the clauses do you mean by the AND, OR buttons?

Yup! The tooltip would be good if it worked on disabled buttons too, whichever way is easier :)

**Pull request merged**